centreon/www/include/configuration/configObject/service/DB-Func.php

```diff
@@ -3723,3 +3733,102 @@ function findHostsOfService(int $serviceId): array
     }
     return $hostIds;
 }
+
+// ------ API Configuration calls --------------------------------------------------------
+
+/**
+ * @param array<int, int> $services The list of service IDs to delete (Ids are the keys)
+ *
+ * @throws Exception
+ */
+function deleteServiceByApi(array $services = []): void
+{
+    global $basePath;
+
+    $serviceIds = array_keys($services);
+    if (empty($serviceIds)) {
+        return;
+    }
+
+    $kernel = Kernel::createForWeb();
+    $router = $kernel->getContainer()->get(Router::class)
+        ?? throw new LogicException('Router not found in container');
+    foreach ($serviceIds as $serviceId) {
+        $url = $router->generate(
+            'DeleteService',
+            ['base_uri' => $basePath, 'serviceId' => $serviceId],
+            UrlGeneratorInterface::ABSOLUTE_URL
+        );
+        callApi($url, 'DELETE', []);
+    }
+}
+
+/**
+ * @param array<int, array> $serviceTemplates The list of service templates to delete (Ids are the keys)
+ *
+ * @throws Exception
+ */
+function deleteServiceTemplateByApi(array $serviceTemplates = []): void
+{
+    global $basePath;
+
+    $serviceTemplateIds = array_keys($serviceTemplates);
+    if (empty($serviceTemplateIds)) {
+        return;
+    }
+
+    $kernel = Kernel::createForWeb();
+    $router = $kernel->getContainer()->get(Router::class)
+        ?? throw new LogicException('Router not found in container');
+    foreach ($serviceTemplateIds as $serviceTemplateId) {
+        $url = $router->generate(
+            'DeleteServiceTemplate',
+            ['base_uri' => $basePath, 'serviceTemplateId' => $serviceTemplateId],
+            UrlGeneratorInterface::ABSOLUTE_URL
+        );
+        callApi($url, 'DELETE', []);
```

Comment:
If a call fails, the deletion process is interrupted. Wouldn't it be better to have the other calls continue and store the status for later display?

Reply:
Indeed, but the current behaviour if you delete multiple services through the UI, and one fail, the process is interrupted so I would like to keep the current behaviour. I'll discuss it with PM to see what we should do !